# Problem 15

he ain't the youngest, he is the second

todo: buy flowers for tomorrow

In [1]:
import datetime, calendar

Let's find the years that start with 1 and end with 6 where the 26th of january was a monday and where there's a leap year (there's feb 29 in the picture!)

In [2]:
possible_years = [year for year in range(1000, 2000) if year % 10 == 6 
                      and datetime.date(year, 1, 26).weekday() == 0
                      and calendar.isleap(year)]
print possible_years

[1176, 1356, 1576, 1756, 1976]


And then let's take the second youngest birth date

In [3]:
print possible_years[-2]

1756


Look for Jan 27, 1756. It's the birth date of Mozart!